### PR TITLE
Feature/indy 121

### PR DESCRIPTION
**Pull request by keenondrums**

### CODE REVIEW

File: sovrin_client/test/cli/test_pool_upgrade_validation.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    validUpgrade[JUSTIFICATION] = ''
+
+    be(trusteeCli)
+    do('send POOL_UPGRADE name={name} version={version} sha256={sha256} '
+       'action={action} schedule={schedule} timeout={timeout} justification={justification}',
+       mapper=validUpgrade, expect=INVALID_SYNTAX, within=10)
+
+
+def testPoolUpgradeHasInvalidSyntaxIfJustificationIsVeryLong(be, do, validUpgrade, trusteeCli):
+    validUpgrade[JUSTIFICATION] = randomString(1001)
```
Comment (lovesh) [inline sovrin_client/test/cli/test_pool_upgrade_validation.py]: Don't hardcode the number 1001, set a constant for max justification size and use a value greater than that

File: sovrin_client/test/cli/test_pool_upgrade_validation.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import pytest
+
+from sovrin_node.test.upgrade.conftest import validUpgrade
+from sovrin_client.test.cli.constants import INVALID_SYNTAX
+from plenum.common.util import randomString
+from sovrin_common.constants import JUSTIFICATION
+
+
+def testPoolUpgradeHasInvalidSyntaxIfJustificationIsEmpty(be, do, validUpgrade, trusteeCli):
+    validUpgrade[JUSTIFICATION] = ''
```
Comment (lovesh) [inline sovrin_client/test/cli/test_pool_upgrade_validation.py]: Is it intentional to keep `JUSTIFICATION` mandatory? If yes then `JUSTIFICATION` should be neither `optional` not `nullable` in `ClientPoolUpgradeOperation`.

Comment (keenondrums) [inline sovrin_client/test/cli/test_pool_upgrade_validation.py]: Nope. Justification is optional, yet if you decided to include it, it must not be empty.

Comment (lovesh) [inline sovrin_client/test/cli/test_pool_upgrade_validation.py]: I get it but doesn't it sound odd? If i don't want to give a justification, i will not include it in txn. So i was asking what is the expectation? Is it mandatory or not?

Comment (keenondrums) [inline sovrin_client/test/cli/test_pool_upgrade_validation.py]: It is not mandatory. It's just a validation for the case when we include justification. I mean you could have a typo or something else and submit an empty justification by mistake.

**Pull request merged**